src/msg/async/ProtocolV2.cc

```diff
@@ -2515,6 +2525,13 @@ CtPtr ProtocolV2::handle_client_ident(char *payload, uint32_t length) {
       client_ident.addrs().front() == entity_addr_t()) {
     return _fault();  // a v2 peer should never do this
   }
+  if (!messenger->get_myaddrs().contains(client_ident.target_addr())) {
```

Comment:
Shouldn’t we do this check only if there is an existing connection already? 

I’m thinking that if we do this check at this point we might be dropping a “good” connection because the client only knows a target_addr with a zero nonce while the server has its own addresses with nonce different than zero.

Reply:
The only case where the client shouldn't have a perfect, exact address is when it is discovering and talking to monitors... which have a zero nonce.